### PR TITLE
JAMES-2586 implement postgres mailbox mapper

### PR DESCRIPTION
**Pull request by hungphan227**

### CODE REVIEW

File: mailbox/postgres/src/test/java/org/apache/james/mailbox/jpa/mail/PostgresMailboxMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        return mailboxMapper;
+    }
+
+    @Override
+    protected MailboxId generateId() {
+        return PostgresMailboxId.generate();
+    }
+
+    @Test
+    void renameShouldUpdateOnyOneMailbox() {
```
Comment (chibenwa) [inline mailbox/postgres/src/test/java/org/apache/james/mailbox/jpa/mail/PostgresMailboxMapperTest.java:60]: renameShouldUpdateOnlyOneMailbox

**Review by chibenwa — COMMENTED** [review]
Please add tests for row level security

**Comment by hungphan227** [comment]
tests for row level security added

File: mailbox/postgres/src/test/java/org/apache/james/mailbox/postgres/mail/PostgresMailboxMapperRowLevelSecurityTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            .create(MailboxPath.forUser(username, "INBOX"), UidValidity.of(1L))
+            .block();
+
+        assertThat(mailboxMapperFactory.getMailboxMapper(session2)
+            .findMailboxByPath(MailboxPath.forUser(username, "INBOX")).block())
+            .isNotNull();
+    }
+
+    @Test
+    void subscriptionsShouldBeIsolatedByDomain() throws Exception {
```
Comment (chibenwa) [inline mailbox/postgres/src/test/java/org/apache/james/mailbox/postgres/mail/PostgresMailboxMapperRowLevelSecurityTest.java:70]: idem...

File: mailbox/postgres/src/test/java/org/apache/james/mailbox/postgres/mail/PostgresMailboxMapperTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        return new PostgresMailboxMapper(new PostgresMailboxDAO(postgresExtension.getPostgresExecutor()));
+    }
+
+    @Override
+    protected MailboxId generateId() {
+        return PostgresMailboxId.generate();
+    }
+
+    @Test
+    void renameShouldUpdateOnlyOneMailbox() {
```
Comment (chibenwa) [inline mailbox/postgres/src/test/java/org/apache/james/mailbox/postgres/mail/PostgresMailboxMapperTest.java:57]: Why is this not in the contract?

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Review by quantranhong1999 — COMMENTED** [review]
I almost forgot: can you do the guice binding for postgres mailbox mapper?

**Review by vttranlina — APPROVED** [review]

**Comment by vttranlina** [comment]
> I almost forgot: can you do the guice binding for postgres mailbox mapper?

Not related but If we guice binding now, the pg-app will not work. 
right now `UidProvider` in postgres mailbox is `JPAUidProvider`, we can not cast PostgresMailboxId to JPAId -> throws an exception

```
java.lang.ClassCastException: class org.apache.james.mailbox.postgres.PostgresMailboxId cannot be cast to class org.apache.james.mailbox.postgres.JPAId (org.apache.james.mailbox.postgres.PostgresMailboxId and org.apache.james.mailbox.postgres.JPAId are in unnamed module of loader 'app')
	at org.apache.james.mailbox.postgres.mail.JPAUidProvider.nextUid(JPAUidProvider.java:71)
```

**Comment by Arsnael** [comment]
@vttranlina Should replace in the code the JPAId cast into PostgresMailboxId then? I could agree to do those bindings and refactoring to an other PR though

**Comment by vttranlina** [comment]
> @vttranlina Should replace in the code the JPAId cast into PostgresMailboxId then? I could agree to do those bindings and refactoring to an other PR though

JPAId based on the LONG type
PostgresMailboxId based on UUID type

We can adapt it, but it not only "cast"

**Comment by Arsnael** [comment]
We gonna have to adapt it if we want to do bindings. It's useless to have a mapper we can't use, I don't see on what we can run perf tests without it :)

**Comment by hungphan227** [comment]
> And drop the JPA mailbox mapper...

for now JPAMailboxMapper should not be removed

**Pull request merged**